src/opf/ptdf.jl

```diff
@@ -0,0 +1,112 @@
+using SparseArrays
+using PowerModels
+using LinearAlgebra
+
+abstract type AbstractPTDF end
+
+
+struct LazyPTDF{TF} <: AbstractPTDF
+    N::Int  # number of buses
+    E::Int  # number of branches
+    islack::Int  # Index of slack bus
+
+    A::SparseMatrixCSC{Float64,Int}  # incidence matrix
+    b::Vector{Float64}  # branch susceptances
+    BA::SparseMatrixCSC{Float64,Int}  # B*A
+    AtBA::SparseMatrixCSC{Float64,Int}  # AᵀBA
+
+    F::LDLt # Factorization of AᵀBA. Must be able to solve linear systems with F \ p
```

Comment:
I believe `LinearAlgebra.LDLt` is for dense factorizations
```suggestion
    F::SparseArrays.CHOLMOD.Factor{Float64, Int64}  # Factorization of AᵀBA
```